src/handshakestate.rs

```diff
@@ -502,6 +502,13 @@ impl HandshakeState {
         self.pattern_position == self.message_patterns.len()
     }
 
+    /// Perform the split calculation and return the resulting keys.
+    pub fn split_raw(&mut self) -> (Vec<u8>, Vec<u8>) {
+        let mut output = ([0u8; MAXHASHLEN], [0u8; MAXHASHLEN]);
+        self.symmetricstate.split_raw(&mut output.0, &mut output.1);
+        (output.0.to_vec(), output.1.to_vec())
```

Comment:
The outputs here are probably not the correct size, since the keys are only `CIPHERKEYLEN` long. I didn't test, but I think the correct version would look like:

```rust
let mut output = (vec![0u8; MAXHASHLEN], vec![0u8; MAXHASHLEN]);
self.symmetricstate.split_raw(&mut output.0, &mut output.1);
output.0.truncate(CIPHERKEYLEN);
output.1.truncate(CIPHERKEYLEN);
output
```